File: app/src/main/java/entities/Vehicles.java
package entities;

import com.google.gson.annotations.SerializedName;

public class Vehicles {

    @SerializedName("id")
    private int id; // this depends on the api, could be something else than integer
    @SerializedName("name")
    private String name;
    @SerializedName("description")
    private String description;
    @SerializedName("latitude")
    private double latitude;
    @SerializedName("longitude")
    private double longitude;
    @SerializedName("batteryLevel")
    private int batteryLevel;
    @SerializedName("timestamp")
    private String timestamp;
    @SerializedName("price")
    private int price;
    @SerializedName("priceTime")
    private int priceTime;
    @SerializedName("currency")
    private String currency;

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public double getLatitude() {
        return latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public int getBatteryLevel() {
        return batteryLevel;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public int getPrice() {
        return price;
    }

    public int getPriceTime() {
        return priceTime;
    }

    public String getCurrency() {
        return currency;
    }
}

File: app/src/main/java/entities/Vehicles.kt
package entities

import com.google.gson.annotations.SerializedName

class Vehicles {

    @SerializedName("id")
    val id: Int = 0 // this depends on the api, could be something else than integer
    @SerializedName("name")
    val name: String? = null
    @SerializedName("description")
    val description: String? = null
    @SerializedName("latitude")
    val latitude: Double = 0.toDouble()
    @SerializedName("longitude")
    val longitude: Double = 0.toDouble()
    @SerializedName("batteryLevel")
    val batteryLevel: Int = 0
    @SerializedName("timestamp")
    val timestamp: String? = null
    @SerializedName("price")
    val price: Int = 0
    @SerializedName("priceTime")
    val priceTime: Int = 0
    @SerializedName("currency")
    val currency: String? = null
}
